File: leoz-node/src/main/java/org/deku/leoz/node/data/entities/master/Sector.java
package org.deku.leoz.node.data.entities.master;

import sx.io.serialization.Serializable;

import javax.persistence.*;
import java.sql.Timestamp;

/**
 * Created by JT on 11.05.15.
 */
@Entity
@Table(name = "mst_sector")
@IdClass(SectorPK.class)
@Serializable(uid = 0xf1dec2bb66db87L )
public class Sector {
    private String sectorFrom;
    private String sectorTo;
    private Timestamp validFrom;

    private Timestamp validTo;
    private String via;
    private Timestamp timestamp;
    private Long syncId;

    @Id
    public String getSectorFrom() {
        return sectorFrom;
    }

    public void setSectorFrom(String sectorFrom) {
        this.sectorFrom = sectorFrom;
    }

    @Id
    public String getSectorTo() {
        return sectorTo;
    }

    public void setSectorTo(String sectorTo) {
        this.sectorTo = sectorTo;
    }

    @Id
    public Timestamp getValidFrom() {
        return validFrom;
    }

    public void setValidFrom(Timestamp validFrom) {
        this.validFrom = validFrom;
    }

    @Basic
    public Timestamp getValidTo() {
        return validTo;
    }

    public void setValidTo(Timestamp validTo) {
        this.validTo = validTo;
    }

    @Basic
    public String getVia() {
        return via;
    }

    public void setVia(String via) {
        this.via = via;
    }

    @Basic
    @Column(nullable = false)
    public Timestamp getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Timestamp timestamp) {
        this.timestamp = timestamp;
    }

    @Basic
    @Column(nullable = false)
    public Long getSyncId() {
        return syncId;
    }

    public void setSyncId(Long syncId) {
        this.syncId = syncId;
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/data/entities/master/Sector.kt
package org.deku.leoz.node.data.entities.master

import sx.io.serialization.Serializable

import javax.persistence.*
import java.sql.Timestamp

/**
 * Created by JT on 11.05.15.
 */
@Entity
@Table(name = "mst_sector")
@IdClass(SectorPK::class)
@Serializable(uid = 0xf1dec2bb66db87L)
class Sector {
    @Id
    var sectorFrom: String? = null
    @Id
    var sectorTo: String? = null
    @Id
    var validFrom: Timestamp? = null
    @Basic
    var validTo: Timestamp? = null
    @Basic
    var via: String? = null
    @Basic
    @Column(nullable = false)
    var timestamp: Timestamp? = null
    @Basic
    @Column(nullable = false)
    var syncId: Long? = null
}
